src/olympia/constants/promoted.py

```diff
@@ -2,9 +2,22 @@
 
 from django.utils.translation import gettext_lazy as _
 
+from olympia.api.utils import APIChoicesWithNone
 from olympia.constants import applications
 
 
+PROMOTED_GROUP_CHOICES = APIChoicesWithNone(
+    ('NOT_PROMOTED', 0, 'Not Promoted'),
```

Comment:
I know `NOT_PROMOTED` is here so it's easier to refactor, but looking at this now -- what would distinguish 'None' from `NOT_PROMOTED`? What was the thought behind `APIChoicesWithNone` over `APIChoices` directly?

IIRC, `NOT_PROMOTED` exists to keep the PromotedAddon around even though the Addon is no longer promoted, but if `None` is an option, wouldn't that be functionally the same?

Reply:
Eventually we will no longer need NOT_PROMOTED because the state is equivalent to the absence of any PromotedAddonPromotions.

That being said, even in that case, there will be no use case for a None value. I'll update it.